packages/platform-ios/src/commands/runIOS/index.js

```diff
@@ -179,6 +182,7 @@ async function runOnSimulator(xcodeProject, args, scheme) {
     args.packager,
     args.verbose,
     args.port,
+    args.terminal,
```

Comment:
Looks like `args.port` and `args.terminal` are passed to all these functions just to call `getProcessOptions` at the very end inside `buildProject`.

Why don't we do:
```js
const processOpts = getProcessOpts(args)
```
here and pass the process options explicitly?

Reply:
@grabbou we call `buildProject` in 2 places. This would cause some unnecessary duplication. 

I think what @lucasbento suggested to combine `args` into single argument of `buildProject` makes sense. It doesn't matter that the order is different between similar functions but feel free to change it to make some more sense, it's only local to this file.

I'm cool with refactoring to this signature:
```js
function buildProject(xcodeProject, udid, scheme, args) {};
```
or generally all functions to something like:
```js
function runOnSimulator(xcodeProject, scheme, args) {};
function runOnDevice(xcodeProject, scheme, args, selectedDevice) {};
function runOnDeviceByUdid(xcodeProject, scheme, args, devices) {};
```
which seams reasonable and I'm happy to have this refactoring as part of this PR :)

Reply:
> Can you elaborate? Do you mean passing args to buildProject and then just passing it along to getProccessOpts?

Yes! The signature proposed by @thymikee is what I had in mind! Just trying to avoid the last few arguments of each of these `run` functions that are not really for `run` function, but for underlying buildProject.

Reply:
Great, I'll work on the `args` refactor.